pkg/config/configuration.go

```diff
@@ -206,3 +276,272 @@ func containsKey(s []string, key string) bool {
+}
+
+// IsOperationAllowedByAccessControlPolicy determines if access control policies allow the operation on the target app
+func IsOperationAllowedByAccessControlPolicy(spiffeID *SpiffeID, srcAppID string, inputOperation string, httpVerb common.HTTPExtension_Verb, accessControlList *AccessControlList) (bool, string) {
+	if accessControlList == nil {
+		// No access control list is provided. Do nothing
+		return isActionAllowed(AllowAccess), ""
+	}
+
+	action := accessControlList.DefaultAction
+	actionPolicy := ActionPolicyGlobal
+
+	if srcAppID == "" {
+		// Did not receive the src app id correctly
+		return isActionAllowed(action), actionPolicy
+	}
+
+	// Look up the src app id in the in-memory table
+	appPolicy, found := accessControlList.PolicySpec[srcAppID]
+
+	if !found {
+		// no policies found for this src app id. Apply global default action
+		return isActionAllowed(action), actionPolicy
+	}
+
+	if spiffeID == nil {
+		// Could not retrieve spiffe id or it is invalid. Apply global default action
+		return isActionAllowed(action), actionPolicy
+	}
+
+	// Match trust domain
+	if appPolicy.TrustDomain != spiffeID.TrustDomain {
+		return isActionAllowed(action), actionPolicy
+	}
+
+	// Match namespace
+	if appPolicy.Namespace != spiffeID.Namespace {
+		return isActionAllowed(action), actionPolicy
+	}
+
+	if appPolicy.DefaultAction != "" {
+		// Since the app has specified a default action, this point onwards,
+		// default action is the default action specified in the spec for the app
+		action = appPolicy.DefaultAction
+		actionPolicy = ActionPolicyApp
+	}
+
+	// the in-memory table has operations stored in the format "/operation name".
+	// Prepend a "/" in case missing so that the match works
+	if !strings.HasPrefix(inputOperation, "/") {
+		inputOperation = "/" + inputOperation
+	}
+
+	inputOperationPrefix, inputOperationPostfix := getOperationPrefixAndPostfix(inputOperation)
+
+	// The acl may specify the operation in a format /invoke/*, get and match only the prefix first
+	operationPolicy, found := appPolicy.AppOperationActions[inputOperationPrefix]
+	if found {
+		// The ACL might have the operation specified as /invoke/*. Here "/*" is stored as the postfix.
+		// Match postfix
+		if operationPolicy.OperationPostFix != "/*" && !strings.HasPrefix(operationPolicy.OperationPostFix, inputOperationPostfix) {
+			return isActionAllowed(action), actionPolicy
+		}
+
+		// Operation prefix and postfix match. Now check the operation specific policy
+		if httpVerb != common.HTTPExtension_NONE {
+			verbAction, found := operationPolicy.VerbAction[httpVerb.String()]
+			if found {
+				// An action for a specific verb is matched
+				action = verbAction
+			} else {
+				verbAction, found = operationPolicy.VerbAction["*"]
+				if found {
+					// The verb matched the wildcard "*"
+					action = verbAction
+				}
+			}
+		} else {
+			// No matching verb found in the operation specific policies.
+			action = appPolicy.DefaultAction
```

Comment:
This is a problem in the case of gRPC, where there will be no verbs.

Reply:
fixed